tests/forcing/test_marrmot.py

```diff
@@ -0,0 +1,123 @@
+                    'variables': {'pr': {'end_year': 1999,
+                                         'mip': 'day',
+                                         'preprocessor': 'daily',
+                                         'start_year': 1989},
+                                  'psl': {'end_year': 1999,
+                                          'mip': 'day',
+                                          'preprocessor': 'daily',
+                                          'start_year': 1989},
+                                  'rsds': {'end_year': 1999,
+                                           'mip': 'day',
+                                           'preprocessor': 'daily',
+                                           'start_year': 1989},
+                                  'rsdt': {'end_year': 1999,
+                                           'mip': 'CFday',
+                                           'preprocessor': 'daily',
+                                           'start_year': 1989},
+                                  'tas': {'end_year': 1999,
+                                          'mip': 'day',
+                                          'preprocessor': 'daily',
+                                          'start_year': 1989}
+                                  }
+
+                }
+            },
+            'documentation': {'authors': ['kalverla_peter',
+                                          'camphuijsen_jaro',
+                                          'alidoost_sarah'],
+                              'projects': ['ewatercycle'],
+                              'references': ['acknow_project']},
+            'preprocessors': {'daily': {'extract_shape': {'crop': True,
+                                                          'method': 'contains',
+                                                          }}}
+        }
+
+    def test_result(self, forcing, tmp_path):
+        expected = MarrmotForcing(
+            directory=str(tmp_path),
+            start_time='1989-01-02T00:00:00Z',
+            end_time='1999-01-02T00:00:00Z',
+            forcing_file='marrmot.mat'
+        )
+        assert forcing == expected
+
+    def test_recipe_configured(self, forcing, mock_recipe_run, reference_recipe, sample_shape):
+        actual = mock_recipe_run['data_during_run']
+        # Remove long description and absolute path so assert is easier
+        actual_desc = actual['documentation']['description']
+        del actual['documentation']['description']
+        actual_shapefile = actual['preprocessors']['daily']['extract_shape']['shapefile']
+        del actual['preprocessors']['daily']['extract_shape']['shapefile']
+
+        assert actual == reference_recipe
+        assert actual_shapefile == sample_shape
+        assert 'MARRMoT' in actual_desc
+
+    def test_saved_yaml(self, forcing, tmp_path):
+        saved_forcing = load(tmp_path)
+
+        assert forcing == saved_forcing
```

Comment:
Should we also add a test for the load_external function?

Reply:
test added